### DISCUSSION TITLE
CSS is broken after upgrading to v11

### DISCUSSION THREAD

**Discussion by mfix22** [vercel/next.js#30593] [Help] [answered]
Hi, I have been unable to upgrade to `next@11` since upgrading breaks the CSS layout, as you can see here in the deployment under this PR: https://github.com/carbon-app/carbon/pull/1249

![image](https://user-images.githubusercontent.com/8397708/139374664-65d35a13-04c4-4d65-a9ee-3986ee03daef.png)

After any re-render (try typing in the main editor), the layout is fixed. 

Cc @samsisle would love your help on this—this issue is preventing us from upgrading, and I would love to try out v12! 🙂

**Comment by samcx** [comment, marked as answer]
@mfix22 I will be taking a look at this soon!

**Reply by mfix22** [reply]
Thanks @samsisle! I really want to use the v11 and v12 features! There is also an open security vulnerability that I can't patch w/o upgrading.

**Reply by samcx** [reply]
@mfix22 I'm afraid there are way too many moving parts to unpack here. I recommend isolating your component with the layout issues out to a separate Next.js repo if possible.

Also, I recommend getting rid of the builds and routes in your `vercel.json` since those very legacy features of Vercel.

**Reply by mfix22** [reply]
I would like to get rid of the `builds` and `routes` from vercel.json, but I am still stuck on the optional params question here: https://github.com/vercel/next.js/discussions/29856#discussioncomment-1592784

**Reply by mfix22** [reply]
@samsisle I believe the issue stems from how this global CSS link is loaded/applied: https://github.com/carbon-app/carbon/blob/main/components/Meta.js#L41 

Can I no longer use CSS this way? Or is there a different place I am required to render links to in the `<head>`?

**Reply by samcx** [reply]
We haven't changed anything in our `<Head>` component as far as I'm aware.

**Reply by mfix22** [reply]
Hmm. It's some weird effect when combining Next.js 11 and Codemirror. The styles get applied when I make an update of any kind, but the initial styles are broken.

**Reply by samcx** [reply]
@mfix22 Took some more time looking at this. Not sure if it's anything related to missing CSS. I do know that the generated css styles here are incorrect. It's also an issue that the dependency hasn't updated in over a year. →

<img width="563" alt="CleanShot 2021-11-28 at 15 56 56@2x" src="https://user-images.githubusercontent.com/28912696/143791563-98e6267c-f39d-43c0-8a29-caffd9ba308c.png">

**Reply by samcx** [reply]
Not sure why this would be affected specifically with Next.js 11, but removing the `container` style fixes the issue. Something definitely up with the styling 

![CleanShot 2021-11-28 at 16 26 45@2x](https://user-images.githubusercontent.com/28912696/143792608-a11ee65e-5f65-4c84-8dcf-3c5307f49e78.png)

**Reply by mfix22** [reply]
@samsisle thanks so much for looking into this again! Are you sure that was the correct `className` to remove? Removing that will completely break the styles of the Carbon editor.

**Reply by mfix22** [reply]
Inlining the nested component containing a `<style jsx>` fixed it.
